### PR TITLE
Support gitscm v3

### PR DESCRIPTION
**Pull request by klimas7**
Resolving issue #217

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -104,7 +104,7 @@
         <dependency>
             <groupId>org.jenkins-ci.plugins</groupId>
             <artifactId>git</artifactId>
-            <version>2.0</version>
+            <version>4.0.0-rc</version>
```
Comment (scaytrase) [inline pom.xml:107]: should we really us 4-RC (not stable) when original issue says about  v3?

Comment (klimas7) [reply pom.xml:107]: Hi,
In issue we have a little mismatch. In the same time was release two plugins:
[git-client](https://plugins.jenkins.io/git-client) 3.0.0-rc version
[git](https://plugins.jenkins.io/git) 4.0.0-rc version

Original issue say about git-client version but this plugin use git plugin.
Unfortunately RC versions was deployed to public Jenkins repository and any client can download that versions. I think if any plugins are deployed to public repository is a stable ;)

Comment (scaytrase) [reply pom.xml:107]: The latest RC for both were released a week ago. I'm not sure it's a good idea to require this as a dependency at the moment.

Also wouldn't it force users to upgrade their plugins to this RC ones?

Comment (klimas7) [reply pom.xml:107]: You're absolutely right, but all user have git plugin and git-client in their Jenkins. Many of them executed jenkins update automatically and got a new versions git and git-client. In that situation stashnotifier plugin doesn't work.

Comment (scaytrase) [reply pom.xml:107]: Sounds legit

**Comment by scaytrase** [comment]
I will build new release as soon as I can

**Comment by scaytrase** [comment]
@klimas7 people got some problems with 4-rc

https://issues.jenkins-ci.org/browse/JENKINS-55939

**Comment by scaytrase** [comment]
Can you verify that https://ci.jenkins.io/blue/organizations/jenkins/Plugins%2Fstashnotifier-plugin/detail/PR-218/1/artifacts HPI works with 4-rc ?

**Comment by klimas7** [comment]
I confirm,  work fine. I checked on this build [#27](https://ci.jenkins.io/job/Plugins/job/stashnotifier-plugin/job/release%252F1.x/27/)

**Comment by scaytrase** [comment]
@klimas7  https://github.com/jenkinsci/stashnotifier-plugin/releases/tag/1.16

**Comment by scaytrase** [comment]
So, we started getting problems, so If we have new features before versions will be stabilized, I'll have to revert this one

https://github.com/jenkinsci/stashnotifier-plugin/issues/221

**Comment by klimas7** [comment]
:( https://github.com/jenkins-infra/update-center2/pull/258 for this situation (I didn't expect that) the best solution will be revert my changes.

**Comment by scaytrase** [comment]
Yeah, looks like we gonna revert

**Pull request merged**